packages/portal/backend/src/server/common/ClasslistAgent.ts

```diff
@@ -0,0 +1,150 @@
+import * as parse from 'csv-parse';
+import * as fs from 'fs';
+import * as rp from "request-promise-native";
+import Config, {ConfigKey} from "../../../../../common/Config";
+import Log from '../../../../../common/Log';
+
+import {ClasslistTransport} from "../../../../../common/types/PortalTypes";
+import {DatabaseController} from "../../controllers/DatabaseController";
+import {PersonController} from "../../controllers/PersonController";
+import {AuditLabel, Grade, Person, PersonKind} from "../../Types";
+import {CSVParser} from "./CSVParser";
+
+export class ClasslistAgent {
+
+    constructor() {
+        //
+    }
+
+    private pc = new PersonController();
+    private db = DatabaseController.getInstance();
+
+    public async fetchClasslist(): Promise<ClasslistTransport[]> {
+        Log.info("IntegrationController::fetchClasslist - start");
```

Comment:
I would update all of these logging messages to refer to `ClasslistAgent::` instead of `IntegrationController::`

Reply:
Agreed.